pkg/appStore/installedApp/read/adapter/adapter.go

```diff
@@ -0,0 +1,76 @@
+package adapter
+
+import (
+	"github.com/devtron-labs/devtron/pkg/appStore/installedApp/read/bean"
+	"github.com/devtron-labs/devtron/pkg/appStore/installedApp/repository"
+	"github.com/devtron-labs/devtron/util"
+)
+
+// GetInstalledAppInternal will return the installed app with environment and cluster details.
+//   - input: installedAppModel *repository.InstalledApps
+//   - returns: *bean.InstalledAppWithEnvAndClusterDetails
+func GetInstalledAppInternal(installedAppModel *repository.InstalledApps) *bean.InstalledAppWithEnvAndClusterDetails {
+	if installedAppModel == nil {
+		return nil
+	}
+	installedAppInternal := &bean.InstalledAppWithEnvAndClusterDetails{}
+	installedAppInternal.InstalledAppMin = util.GetDeReferencedBean(GetInstalledAppMin(installedAppModel))
+	// Extra App details
+	if !installedAppModel.App.IsZero() {
+		installedAppInternal.AppName = installedAppModel.App.AppName
+		installedAppInternal.AppOfferingMode = installedAppModel.App.AppOfferingMode
+		installedAppInternal.TeamId = installedAppModel.App.TeamId
+	}
+	// Extra Environment details
+	if !installedAppModel.Environment.IsZero() {
+		installedAppInternal.EnvironmentName = installedAppModel.Environment.Name
+		installedAppInternal.EnvironmentIdentifier = installedAppModel.Environment.EnvironmentIdentifier
+		installedAppInternal.Namespace = installedAppModel.Environment.Namespace
+		installedAppInternal.ClusterId = installedAppModel.Environment.ClusterId
+	}
+	// Cluster details
+	if !installedAppModel.Environment.Cluster.IsZero() {
+		installedAppInternal.ClusterName = installedAppModel.Environment.Cluster.ClusterName
+	}
```

Comment:
please double check if any other fields were used earlier

Reply:
as I have replaced the response type from `installedAppModel` to `installedAppInternal`. So I've also updated the usage of installedAppModel and added the required fields to installedAppInternal struct.